editor/src/components/canvas/context-menu-state.tsx

```diff
@@ -0,0 +1,41 @@
+import { atom, useAtom } from 'jotai'
+
+import { BodyMenuOpenClass } from '../../core/shared/utils'
+import React from 'react'
+
+export const ContextMenuStateAtom = atom([] as string[])
+export const ContextMenuOpen = atom((get) => {
+  const contextMenusOpen = get(ContextMenuStateAtom)
+  return contextMenusOpen.length > 0
+})
+
+export const useContextMenuState = () => {
+  const [values, setValues] = useAtom(ContextMenuStateAtom)
+  const [isContextMenuOpen] = useAtom(ContextMenuOpen)
+
+  React.useEffect(() => {
+    if (isContextMenuOpen) {
+      document.body.classList.add(BodyMenuOpenClass)
+    } else {
+      document.body.classList.remove(BodyMenuOpenClass)
+    }
+  }, [isContextMenuOpen])
+
+  const add = (id: string) => {
+    if (values.includes(id)) return
```

Comment:
style - I think the convention is for always using if blocks (instead of a single line)

Reply:
We should put this stuff in eslint 🙈